sdk/core/Microsoft.Extensions.Azure/src/AzureClientBuilderExtensions.cs

```diff
@@ -10,43 +10,103 @@
             return builder;
         }
 
+        /// <summary>
+        /// Adds a delegate to configure the client options. All delegates are executed in order they were added.
+        /// </summary>
+        /// <typeparam name="TClient">The type of the client.</typeparam>
+        /// <typeparam name="TOptions">The options type the client uses.</typeparam>
+        /// <param name="builder">The client builder instance.</param>
+        /// <param name="configureOptions">The delegate to use to configure options.</param>
+        /// <returns>The client builder instance.</returns>
         public static IAzureClientBuilder<TClient, TOptions> ConfigureOptions<TClient, TOptions>(this IAzureClientBuilder<TClient, TOptions> builder, Action<TOptions> configureOptions) where TOptions : class
         {
             return builder.ConfigureOptions((options, _) => configureOptions(options));
         }
 
+        /// <summary>
+        /// Configures client options using provided <see cref="IConfiguration"/> instance.
+        /// </summary>
+        /// <typeparam name="TClient">The type of the client.</typeparam>
+        /// <typeparam name="TOptions">The options type the client uses.</typeparam>
+        /// <param name="builder">The client builder instance.</param>
+        /// <param name="configuration">The configuration instance to use.</param>
+        /// <returns>The client builder instance.</returns>
         public static IAzureClientBuilder<TClient, TOptions> ConfigureOptions<TClient, TOptions>(this IAzureClientBuilder<TClient, TOptions> builder, IConfiguration configuration) where TOptions : class
         {
             return builder.ConfigureOptions(options => configuration.Bind(options));
         }
 
+        /// <summary>
+        /// Adds a delegate to configure the client options. All delegates are executed in order they were added.
+        /// </summary>
+        /// <typeparam name="TClient">The type of the client.</typeparam>
+        /// <typeparam name="TOptions">The options type the client uses.</typeparam>
+        /// <param name="builder">The client builder instance.</param>
+        /// <param name="configureOptions">The delegate to use to configure options.</param>
+        /// <returns>The client builder instance.</returns>
         public static IAzureClientBuilder<TClient, TOptions> ConfigureOptions<TClient, TOptions>(this IAzureClientBuilder<TClient, TOptions> builder, Action<TOptions, IServiceProvider> configureOptions) where TOptions : class
         {
             var impl = builder.ToBuilder();
             impl.ServiceCollection.AddSingleton<IConfigureOptions<TOptions>>(provider => new ConfigureClientOptions<TClient,TOptions>(provider, impl.Registration, configureOptions));;
             return builder;
         }
 
+        /// <summary>
+        /// Sets the service version to use for this client registration. Latest by default.
```

Comment:
"Latest by default" is a little tricky.  It's `Latest` if you don't call `WithVersion` but `WithVersion(default(TOptions.ServiceVersion))` would not give you Latest for Storage.